src/scaffoldmaker/utils/tubemesh.py

```diff
@@ -281,71 +280,87 @@ def getCoordinatesFromInner(xInner, d1Inner, d2Inner, d3Inner,
         relativeThicknessList.append(relativeThicknessList[-1])
 
     for n2 in range(elementsCountAlong + 1):
-        wallThickness = wallThicknessList[n2]
+        if outward:
+            wallThickness = wallThicknessList[n2]
+        else:
+            wallThickness = -wallThicknessList[n2]
```

Comment:
The code below has several near identical cases for inward and outward which are different only because of the above sign. Don't modify the wallThickness, instead have a new variable:
```
wallOutwardDisplacement = wallThickness if outward else -wallThickness
```
Use it in the extrusion code below (see other comment).

Reply:
Done.